### PR TITLE
Add support for sign op

### PR DESCRIPTION
**Pull request by ajakovljevicTT**
- Add end-to-end implementation of the logp1 op
 - Add stablehlo to ttir to ttnn IR conversion of the op
 - Related to issue https://github.com/tenstorrent/tt-mlir/issues/1001

### CODE REVIEW

File: test/ttmlir/Silicon/TTNN/simple_eltwise.mlir [hunk 1]
```diff
@@ -134,3 +134,11 @@ func.func @maximum(%arg0: tensor<64x128xf32>, %arg1: tensor<64x128xf32>) -> tens
...
   return %1 : tensor<64x128xf32>
 }
+
+func.func @sign(%arg0: tensor<64x128xf32>) -> tensor<64x128xf32> {
+  // CHECK: %[[C:.*]] = "ttnn.empty"[[C:.*]]
+  %0 = tensor.empty() : tensor<64x128xf32>
+  // CHECK: %[[C:.*]] = "ttnn.sign"[[C:.*]]
+  %1 = "ttir.sign"(%arg0, %0) <{operandSegmentSizes = array<i32: 1, 1>, operand_constraints = [#any_device, #any_device]}> : (tensor<64x128xf32>, tensor<64x128xf32>) -> tensor<64x128xf32>
+  return %1 : tensor<64x128xf32>
+}
```
Comment (svuckovicTT) [inline test/ttmlir/Silicon/TTNN/simple_eltwise.mlir:144]: Missing newline?

**Review by svuckovicTT — APPROVED** [review]

File: test/ttmlir/Silicon/TTNN/simple_eltwise.mlir (hunk 1 above)
Comment (ajakovljevicTT) [reply test/ttmlir/Silicon/TTNN/simple_eltwise.mlir]: It has now been fixed.

**Review by jnie-TT — APPROVED** [review]

File: test/ttmlir/Conversion/StableHLOToTTIR/sign_op.mlir [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// REQUIRES: stablehlo
+// RUN: ttmlir-opt --stablehlo-to-ttir-pipeline %s | FileCheck %s
+#any_device = #tt.operand_constraint<dram|l1|scalar|tile|any_device|any_device_tile>
+module @jit_eltwise_sign attributes {} {
+  func.func public @test_sign(%arg0: tensor<13x21x3xf32>) -> tensor<13x21x3xf32> {
+    %0 = stablehlo.sign %arg0 : tensor<13x21x3xf32>
+    // CHECK: %[[C:.*]] = tensor.empty[[C:.*]]
```
Comment (mrakitaTT) [inline test/ttmlir/Conversion/StableHLOToTTIR/sign_op.mlir:7]: Can you please improve these checks in tests that you've added? See this message: https://tenstorrent.slack.com/archives/C07A30HRX7C/p1727698795548319

Comment (ajakovljevicTT) [reply test/ttmlir/Conversion/StableHLOToTTIR/sign_op.mlir]: Noted. The more detailed checks in tests have been added in this and other mentioned PRs.

File: test/ttmlir/Dialect/TTNN/eltwise/unary/sign/simple_sign.mlir [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline %s | FileCheck %s
+#any_device = #tt.operand_constraint<dram|l1|scalar|tile|any_device|any_device_tile>
+module attributes {} {
+  func.func @forward(%arg0: tensor<64x128xf32>) -> tensor<64x128xf32> {
+    %0 = tensor.empty() : tensor<64x128xf32>
+    // CHECK: [[VAL0:%[0-9]+]] = "ttnn.empty"(%{{[0-9]+}}) <{dtype = {{.*}}, layout = {{.*}}, memory_config = {{.*}}, <{{.*}}>>, shape = #ttnn.shape<[[TENSOR_SHAPE:[0-9]+x[0-9]+]]>}>
+    %1 = "ttir.sign"(%arg0, %0) <{operandSegmentSizes = array<i32: 1, 1>, operand_constraints = [#any_device, #any_device]}> : (tensor<64x128xf32>, tensor<64x128xf32>) -> tensor<64x128xf32>
+    // CHECK: %{{[0-9]+}} = "ttnn.sign"(%{{[0-9]+}}, [[VAL0]]) <{operandSegmentSizes = array<i32: 1, 1>}> : (tensor<[[TENSOR_SHAPE]]x{{.*}}, {{.*}}>, tensor<[[TENSOR_SHAPE]]x{{.*}}, {{.*}}) -> tensor<[[TENSOR_SHAPE]]x{{.*}}, {{.*}}>
```
Comment (mrakitaTT) [inline test/ttmlir/Dialect/TTNN/eltwise/unary/sign/simple_sign.mlir:8]: You can capture this into VAL1 to use with return check

File: test/ttmlir/Silicon/TTNN/simple_eltwise.mlir [hunk 4]
```diff
@@ -134,3 +134,12 @@ func.func @maximum(%arg0: tensor<64x128xf32>, %arg1: tensor<64x128xf32>) -> tens
   %1 = "ttir.maximum"(%arg0, %arg1, %0) <{operandSegmentSizes = array<i32: 2, 1>, operand_constraints = [#any_device, #any_device, #any_device]}> : (tensor<64x128xf32>, tensor<64x128xf32>, tensor<64x128xf32>) -> tensor<64x128xf32>
   return %1 : tensor<64x128xf32>
 }
+
+func.func @sign(%arg0: tensor<64x128xf32>) -> tensor<64x128xf32> {
+  %0 = tensor.empty() : tensor<64x128xf32>
+  // CHECK: [[VAL0:%[0-9]+]] = "ttnn.empty"(%{{[0-9]+}}) <{dtype = {{.*}}, layout = {{.*}}, memory_config = {{.*}}, <{{.*}}>>, shape = #ttnn.shape<[[TENSOR_SHAPE:[0-9]+x[0-9]+]]>}>
+  %1 = "ttir.sign"(%arg0, %0) <{operandSegmentSizes = array<i32: 1, 1>, operand_constraints = [#any_device, #any_device]}> : (tensor<64x128xf32>, tensor<64x128xf32>) -> tensor<64x128xf32>
+  // CHECK: %{{[0-9]+}} = "ttnn.sign"(%{{[0-9]+}}, [[VAL0]]) <{operandSegmentSizes = array<i32: 1, 1>}> : (tensor<[[TENSOR_SHAPE]]x{{.*}}, {{.*}}>, tensor<[[TENSOR_SHAPE]]x{{.*}}, {{.*}}) -> tensor<[[TENSOR_SHAPE]]x{{.*}}, {{.*}}>
```
Comment (mrakitaTT) [inline test/ttmlir/Silicon/TTNN/simple_eltwise.mlir:142]: You can capture this into VAL1 to use with return check

**Review by mrakitaTT — APPROVED** [review]
Thanks for writing detailed tests!

File: test/ttmlir/Dialect/TTNN/eltwise/unary/sign/simple_sign.mlir (hunk 3 above)
Comment (ajakovljevicTT) [reply test/ttmlir/Dialect/TTNN/eltwise/unary/sign/simple_sign.mlir:8]: As the return argument is different from the result of the sign operation (due to the "ttnn.to_memory_config" operation in between), I did not capture the result to be used with the return check. Example of translated code is below.
```
    %4 = "ttnn.sign"(%2, %3) <{operandSegmentSizes = array<i32: 1, 1>}> : (tensor<64x128xf32, #layout1>, tensor<64x128xf32, #layout1>) -> tensor<64x128xf32, #layout1>
    "ttnn.dealloc"(%2) : (tensor<64x128xf32, #layout1>) -> ()
    %5 = "ttnn.to_memory_config"(%4, %0) : (tensor<64x128xf32, #layout1>, !tt.device<#device>) -> tensor<64x128xf32, #layout>
    "ttnn.dealloc"(%3) : (tensor<64x128xf32, #layout1>) -> ()
    return %5 : tensor<64x128xf32, #layout>
```

Comment (ajakovljevicTT) [reply test/ttmlir/Dialect/TTNN/eltwise/unary/sign/simple_sign.mlir:8]: As the return argument is different from the result of the sign operation (due to the "ttnn.to_memory_config" operation in between), I did not capture the result to be used with the return check. Example of translated code is below.
```
    %4 = "ttnn.sign"(%2, %3) <{operandSegmentSizes = array<i32: 1, 1>}> : (tensor<64x128xf32, #layout1>, tensor<64x128xf32, #layout1>) -> tensor<64x128xf32, #layout1>
    "ttnn.dealloc"(%2) : (tensor<64x128xf32, #layout1>) -> ()
    %5 = "ttnn.to_memory_config"(%4, %0) : (tensor<64x128xf32, #layout1>, !tt.device<#device>) -> tensor<64x128xf32, #layout>
    "ttnn.dealloc"(%3) : (tensor<64x128xf32, #layout1>) -> ()
    return %5 : tensor<64x128xf32, #layout>
```

File: test/ttmlir/Silicon/TTNN/simple_eltwise.mlir (hunk 4 above)
Comment (ajakovljevicTT) [reply test/ttmlir/Silicon/TTNN/simple_eltwise.mlir:142]: Same comment as above.

**Review by tapspatel — APPROVED** [review]

File: test/ttmlir/Dialect/TTNN/eltwise/unary/sign/simple_sign.mlir (hunk 3 above)
Comment (mrakitaTT) [reply test/ttmlir/Dialect/TTNN/eltwise/unary/sign/simple_sign.mlir:8]: Oh right, I forgot about that. All good then.

**Review by sdjordjevicTT — APPROVED** [review]

**Review by tapspatel — CHANGES_REQUESTED** [review]
do you mind adding a sign test under Silicon/TTNN/perf_unit? We added some perf support yesterday (other than that looks awesome)

**Review by tapspatel — APPROVED** [review]

**Pull request merged**